docs/DeveloperGuide.adoc

```diff
@@ -295,72 +319,395 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+* 2b.
+If just typing the index, the lion returns the total profit for that item
+
+=== Cashier Tab
+
+[discrete]
+=== Use Case 10: Stores the information of transactions
+
+*Guarantees*
+
+- Transaction is valid only if the quantity in the inventory is more than or equal to the quantity keyed in.
+- The type of item bought is already listed in the inventory
+
+*MSS*
+
+1. The user arrives on the ‘Cashier’ tab.
+2. The user chooses to update the purchases made.
+3. CS requests for details of the purchase.
+4. User enters the requested details.
+5. CS process the purchase and displays the amount change, if valid.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 4a.
+CS detects that the item is not listed in the inventory or quantity is insufficient
++
+[none]
+** 4a1. CS requests for valid item
+** 4a2. User enters new item.
+** 4a3. Steps 4a1 to 4a2 are repeated until data entered are correct.
+** 4a4. Use case resumes from Step 5.
+
+* *a.
+At any time, User chooses to cancel the purchase made.
++
+[none]
+** *a1. CS requests for confirmation.
+** *a2. User confirms the cancellation.
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
 
-.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+. Technical Requirements:
+* Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
+* Should work on both 32-bit and 64-bit environments.
+* Should be easy to navigate to other tabs
+. Quality Requirements:
+* Should be easy to pick up and intuitive for novice user
+* A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+* The messages the lion says should be understandable and not result in information overload
+* The GUI should look neat and organised
+* Should be intuitive and easy to understand and learn for a total novice
+* Our mascot should be original or not copied
+* Clear and grammatically correct English should be used throughout the program
+. Process Requirements:
+* Members should aim to do increments according to schedule
+* Members should sound out when help is needed in their increments
+* Members will be doing back-end first before proceeding to front-end designing
+. Data Requirements:
+* Should be able to store sufficient data for 1 year’s worth of transactions within a CCA
+* Should be as much resistant to data loss as possible
+* Should be able to read data quickly to prevent long loading times
 
 [appendix]
 == Glossary
```

Comment:
Would be good to include the definitions of the subjects in the user stories, e.g. members, new users.